From df87420725157953268ed099c3c97989288db1fa Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <fsumsal@redhat.com>
Date: Wed, 13 Mar 2024 12:13:23 +0100
Subject: [PATCH] ci: reduce ASLR entropy

The latest GH Action runners started using 32-bit entropy for ASLR,
which makes it incompatible with llvm-14. This was fixed in later llvm
releases, but these aren't available on Ubuntu Jammy (22.04). Let's
reduce the ASLR entropy to 28-bit, which should make llvm happy again,
until the issue is resolved.

See: actions/runner-images#9491
---
 .github/workflows/unit_tests.yml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/.github/workflows/unit_tests.yml b/.github/workflows/unit_tests.yml
index f397e8ed6e..814e17b6bf 100644
--- a/.github/workflows/unit_tests.yml
+++ b/.github/workflows/unit_tests.yml
@@ -18,6 +18,9 @@ jobs:
     steps:
       - name: Repository checkout
         uses: actions/checkout@v1
+      # FIXME: drop once https://github.com/actions/runner-images/issues/9491  is resolved
+      - name: Reduce ASLR entropy
+        run: sudo sysctl -w vm.mmap_rnd_bits=28
       - name: Install build dependencies
         run: sudo -E .github/workflows/unit_tests.sh SETUP
       - name: Build & test (${{ env.CENTOS_RELEASE }} / ${{ matrix.phase }})
